File: MediaServiceCore/youtubeapi/src/main/java/com/liskovsoft/youtubeapi/next/v1/WatchNextApiHelper.java
package com.liskovsoft.youtubeapi.next.v1;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import com.liskovsoft.youtubeapi.common.helpers.PostDataHelper;

public class WatchNextApiHelper {
    public static String getWatchNextQuery(@NonNull String videoId) {
        return getWatchNextQuery(videoId, null, 0);
    }

    public static String getWatchNextQuery(@Nullable String videoId, @Nullable String playlistId, int playlistIndex) {
        return getWatchNextQuery(videoId, playlistId, playlistIndex, null);
    }

     
    public static String getWatchNextQuery(@Nullable String videoId, @Nullable String playlistId, int playlistIndex, String playlistParams) {
         
        String videoData = String.format("\"videoId\":\"%s\"", videoId);

         
         
        if (playlistId != null) {
            videoData += String.format(",\"playlistId\":\"%s\",\"playlistIndex\":%s", playlistId, Math.max(playlistIndex, 0));
        }

        if (playlistParams != null) {
            videoData += String.format(",\"params\":\"%s\"", playlistParams);
        }

        return PostDataHelper.createQueryTV(videoData);
    }
}

File: MediaServiceCore/youtubeapi/src/main/java/com/liskovsoft/youtubeapi/next/v2/WatchNextApiHelper.kt
package com.liskovsoft.youtubeapi.next.v2

import com.liskovsoft.youtubeapi.common.helpers.AppClient
import com.liskovsoft.youtubeapi.common.helpers.PostDataType
import com.liskovsoft.youtubeapi.common.helpers.QueryBuilder

internal object WatchNextApiHelper {
    fun getWatchNextQuery(client: AppClient, videoId: String): String {
        return getWatchNextQuery(client, videoId, null, 0)
    }

    fun getWatchNextQuery(appClient: AppClient, videoId: String?, playlistId: String?, playlistIndex: Int): String {
        return getWatchNextQuery(appClient, videoId, playlistId, playlistIndex, null)
    }

     
    fun getWatchNextQuery(client: AppClient, videoId: String?, playlistId: String?, playlistIndex: Int, playlistParams: String?): String {
        return QueryBuilder(client)
            .setType(PostDataType.Browse)
            .setVideoId(videoId ?: "")  
            .setPlaylistId(playlistId)
            .setPlaylistIndex(playlistIndex)
            .setParams(playlistParams)  
            .build()
    }

    fun getUnlocalizedTitleQuery(videoId: String): String {
        return "https://www.youtube.com/watch?v=$videoId"
    }
}

